src/btf.h

```diff
@@ -70,18 +70,16 @@ class BTF {
   };
 
 public:
-  BTF(const std::set<std::string>& modules);
-  BTF(BPFtrace* bpftrace, const std::set<std::string>& modules) : BTF(modules)
-  {
-    bpftrace_ = bpftrace;
-  };
+  BTF(BPFtrace* bpftrace);
```

Comment:
Nit: Since this is a cleanup, can this be decomposed into what `BTF` actually needs? Right now you have:

```
Bpftrace <----> BTF
```

Which suggests that there's some composition problem, and ideally you have:

```
        Bpftrace ------> BTF
             \          /
              \        /
               v      v
             SomethingData
```

As a general rule, I think that the hierarchy shouldn't have cycles unless you need to.

Reply:
I know but that is a larger change. It references bpftrace's bpffeature object, which is fine; I can pass that as a dependency but it also updates bpftrace's structs and calls some of its helper functions.